shared/editor/queries/getRowIndex.ts

```diff
@@ -2,10 +2,17 @@ import { CellSelection } from "prosemirror-tables";
 
 export default function getRowIndex(selection: CellSelection) {
   const isRowSelection = selection.isRowSelection && selection.isRowSelection();
-  if (!isRowSelection) {
-    return undefined;
+  if (isRowSelection) {
+    const path = (selection.$from as any).path;
+    return path[path.length - 8];
+  } else {
+    const fromStr = selection.$from.toString();
```

Comment:
I've honestly never used this method, not sure what the format of toString looks like but it's likely that as it's just stringifying the internal selection structure there is a better way of pulling the data out?

I'd also want to check that modifying the behavior of this method does not change how the selection toolbar works inside tables, does text selection in a table cell still show the correct toolbar?

Reply:
You were right about both.

The toString wasn't ideal, I didn't quite understand how ProseMirror worked. I'm now using the same method you were using, just different index in the array (`path[path.length - 11]`).

To avoid changing the toolbar I created a separate function in the same `getRowIndex` file. Both methods get the row index so I figured it was ok to leave them in the same file. Let me know if it needs a different structure.